src/main/java/org/example/movieSpring/service/MovieService.java

```diff
@@ -0,0 +1,51 @@
+package org.example.movieSpring.service;
+
+import org.example.movieSpring.model.Movie;
+import org.example.movieSpring.repo.MovieRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieService {
+
+    private final MovieRepo movieRepo;
+
+    @Autowired
+    public MovieService(MovieRepo movieRepo) {
+        this.movieRepo = movieRepo;
+    }
+
+    public List<Movie> getMovies() {
+        return movieRepo.findAll();
+    }
+
+    public Optional<Movie> getMovieById(Long id) {
+        return movieRepo.findById(id);
+    }
+
+    public Movie addMovie(Movie movie) {
+        return movieRepo.save(movie);
+    }
```

Comment:
what happens when someone sends a movie that already exists?
often update and save are combined in one method with some logic deciding which one to pick

Reply:
I will add functionality that searches through the existing DB and sees if any of the names are taken (will make it fuzzy) and if it exists, return some info that it already exists

" often update and save are combined in one method with some logic deciding which one to pick"
I will fix this too